addons/stock/models/stock_location.py

```diff
@@ -501,10 +501,14 @@ def _onchange_warehouse_selectable(self):
         if not self.warehouse_selectable:
             self.warehouse_ids = [(5, 0, 0)]
 
-    def toggle_active(self):
-        for route in self:
-            route.with_context(active_test=False).rule_ids.filtered(lambda ru: ru.active == route.active).toggle_active()
-        super().toggle_active()
+    def write(self, vals):
+        if 'active' in vals:
+            rules = self.with_context(active_test=False).rule_ids.sudo().filtered(lambda rule: rule.location_dest_id.active)
+            if vals['active']:
+                rules.action_unarchive()
+            else:
+                rules.action_archive()
+        return super().write(vals)
```

Comment:
`toggle_active` was replace by `write` in [saas-18.1](https://github.com/odoo/odoo/pull/183691).
To have minimal change in stables, it is better to stick to `toggle_active` until then
```suggestion
    def toggle_active(self):
        for route in self:
            route.with_context(active_test=False).rule_ids.sudo().filtered(lambda ru: ru.location_dest_id.active and ru.active == route.active).toggle_active()
        super().toggle_active()
```

Reply:
Done.